### PR TITLE
(feat) Introduced Avatars for players

### LINKED ISSUE #68: Introduce Avatars for players to choose in addtion to nicknames

### PR DESCRIPTION
**Pull request by existentialcoder**
Fixes #68 

### Description
Introduced avatars from players

### Motivation and Context
Avatars are a good to have for a pictionary game

### How Has This Been Tested?
Tested in local with multi tabs

### Screenshots, if any
Screenshot from 2020-09-15 22-25-20
Screenshot from 2020-09-15 22-12-54

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by mahadevans87** [comment]
Hey This looks awesome! Can't wait to try this out. will review this sometime tomorrow. Thanks

File: client/src/components/dialogs/AvatarPicker.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import './avatar-picker.css';
+
+class AvatarPicker extends React.Component {
```
Comment (mahadevans87) [inline client/src/components/dialogs/AvatarPicker.js:4]: @existentialcoder Please try to use Functional components as much as possible as it goes with the current style of code + React also recommends the same.

File: server/controller/index.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    const rand = Math.floor(Math.random() * (length + 1));
+    const temp = arr[rand];
+    arr[rand] = arr[length];
+    arr[length] = temp;
+  }
+  return arr;
+};
+
+const avatarController = () => {
+  const allFiles = getAllFiles(path.join(process.cwd(), '/client/src/assets/avatars'))
```
Comment (mahadevans87) [inline server/controller/index.js:33]: The path /client/src means that the server and the client has to be on the same instance right. We may want to scale the frontend and backend independently in the future. For e.g the React app could be hosted on an S3 bucket or Netlify.

How about just sending the avatar ids from the server to client and the client uses local images to map it to the actual image?

**Review by mahadevans87 — CHANGES_REQUESTED** [review]
@existentialcoder Really sorry for the late review. The PR looks exciting. I have a couple of queries that I have attached in the PR. Please take a look and let me know what you think

**Comment by mahadevans87** [comment]
@existentialcoder Do you think you can fix the changes and submit a PR sometime over the next week?

File: client/src/components/dialogs/AvatarPicker.js (hunk 1 above)
Comment (existentialcoder) [reply client/src/components/dialogs/AvatarPicker.js:4]: In functional components I faced an infinite loop problem! - https://stackoverflow.com/questions/59112809/react-usestate-hook-causing-infinite-loop

File: server/controller/index.js (hunk 2 above)
Comment (existentialcoder) [reply server/controller/index.js:33]: Yes :+1:

**Comment by existentialcoder** [comment]
> @existentialcoder Do you think you can fix the changes and submit a PR sometime over the next week?

Sure

File: client/src/components/dialogs/AvatarPicker.js (hunk 1 above)
Comment (sampathBlam) [reply client/src/components/dialogs/AvatarPicker.js]: @mahadevans87 : Changed it to functional components. 
@existentialcoder : Nice catch. Like it is mentioned in the above stackoverflow link, we have to check the avatars array before hitting the URL. If it is present, we need not fetch it again, hence preventing a re-render.

**Comment by sampathBlam** [comment]
Merging the changes for avatar.

**Pull request merged**